chain/network/src/peer_manager.rs

```diff
@@ -376,8 +396,11 @@ impl PeerManagerActor {
     }
 
     /// Get a random peer we are not connected to from the known list.
-    fn sample_random_peer(&self, ignore_list: &HashSet<PeerId>) -> Option<PeerInfo> {
-        let unconnected_peers = self.peer_store.unconnected_peers(ignore_list);
+    fn sample_random_peer(&self) -> Option<PeerInfo> {
+        let unconnected_peers = self.peer_store.unconnected_peers(|peer_id| {
+            // Ignore peers in active set and in outgoing list
+            self.active_peers.contains_key(&peer_id) || self.outgoing_peers.contains(&peer_id)
```

Comment:
why can a peer be both unconnected and active?

Reply:
It can't. This statement here is to avoid connecting to peers that we are already connected (`active_pweers`) and to peers we already initiated a connection to (`outgoing_peers`)

Reply:
This is what `unconnected_peers` does before
```
pub fn unconnected_peers(&self, ignore_list: &HashSet<PeerId>) -> Vec<PeerInfo> {
        self.find_peers(
            |p| {
                (p.status == KnownPeerStatus::NotConnected || p.status == KnownPeerStatus::Unknown)
                    && !ignore_list.contains(&p.peer_info.id)
                    && p.peer_info.addr.is_some()
            },
            0,
        )
    }
```
It seems that if peer status is properly maintained then we don't need this filtering. Why is peer status not in sync with the actual connections? For example, if a peer is in active peers, I would expect it to have `KnownPeerStatus::Connected`

Reply:
Totally miss that, you are right. Will revert this change.